### DISCUSSION TITLE
Does x32dbg Support Source Level Debugging of DLLs Compiled With Visual Studio Express 2010?

### DISCUSSION THREAD

**Discussion by hankinsohl** [x64dbg/x64dbg#3433] [Q&A]
I'm working on an older codebase and need to use Visual Studio Express 2010 (VSE).  Using VSE I've built a debug version of the DLL and within the VSE IDE I can source-level debug the DLL.  I'd like to source level debug the DLL within x32dbg if this is possible.  Using x32dbg I can attach to the program that uses the DLL and looking at the symbols window I see that symbols for the DLL are loaded.  I don't see any options though to step into the source code for the DLL.

Am I missing something or is source level debugging of such DLLs not supported?

**Comment by mrexodia** [comment]
Right click on the disassembly -> open source. That being said, source debugging isn’t really supported. You can step through the lines and see the source, but you won’t see variables etc.

**Comment by hankinsohl** [comment]
Thanks for the response.  Even though I knew that the drop down menu was supposed to offer an option to view the source code I somehow missed it every time I tried to look at the source.  Anyhow, open source did work.

**Reply by mrexodia** [reply]
You can also enable this option to automatically open the source code when stepping around:

![image](https://github.com/user-attachments/assets/9853523b-f2a7-4302-a495-f4017fd1d10b)

**Comment by hankinsohl** [comment]
Thanks!  Nice to know.
